contracts/farm-bot.sol

```diff
@@ -131,52 +136,53 @@ contract FarmBot is Owned, FarmbotERC20 {
         stakingRewards.stake(tokenBalance);
     }
 
-    function claimRewards(uint deadline) public ensure(deadline) {
-        stakingRewards.getReward();
-        uint256 tokenBalance = rewardsToken.balanceOf(address(this));
-
-        if (tokenBalance == 0) {
-            return;
-        }
+    // convenience method for anyone considering calling claimRewards (who may want to compare bounty to gas cost)
+    function previewBounty() external view returns (TokenAmount[] memory) {
+        uint _leftoverBalance = rewardsToken.balanceOf(address(this));
+        uint _interestEarned = stakingRewards.earned(address(this));
+        TokenAmount[] memory _rewardsTokenBalances = new TokenAmount[](1);
+        _rewardsTokenBalances[0] = TokenAmount(rewardsToken, _interestEarned + _leftoverBalance);
+        return revoBounty.calculateFeeBounty(_rewardsTokenBalances);
+    }
 
-        uint256 feeAmount = tokenBalance * feeNumerator / feeDenominator;
-        uint256 halfTokens = (tokenBalance - feeAmount) / 2;
-
-        uint256 amountToken0;
-        // Figure out best-case scenario amount of token0 we can get and swap
-        if (path0.length >= 2) {
-            uint[] memory amountsOut = router.getAmountsOut(halfTokens, path0);
-            uint amountOut = amountsOut[amountsOut.length - 1];
-            rewardsToken.approve(address(router), halfTokens);
-            uint[] memory amountsSwapped = router.swapExactTokensForTokens(
-                halfTokens,
-                amountOut * slippageNumerator / slippageDenominator,
-                path0,
+    // Figure out best-case scenario amount of token we can get and swap
+    function swapForTokenInPool(address[] storage _swapPath, uint _startTokenBudget, IERC20 _startToken, uint _deadline) private returns (uint256) {
```

Comment:
this was needed to fix a compilation error related to there being too many local variables in `claimRewards`. Advice on eth stack exchange was to break the function down into helpers, and this looked like the best option for that